kroxylicious-filter-test-support/src/main/java/io/kroxylicious/test/record/RecordTestUtils.java

```diff
@@ -295,7 +308,54 @@ public static MemoryRecords memoryRecords(byte magic, long offset, long timestam
     }
 
     private static MemoryRecords memoryRecordsWithoutCopy(byte magic, long offset, long timestamp, byte[] key, byte[] value, Header... headers) {
-        try (MemoryRecordsBuilder memoryRecordsBuilder = new MemoryRecordsBuilder(
+        try (MemoryRecordsBuilder memoryRecordsBuilder = defaultMemoryRecordsBuilder(magic)) {
+            memoryRecordsBuilder.appendWithOffset(offset, timestamp, key, value, headers);
+            return memoryRecordsBuilder.build();
+        }
+    }
+
+    /**
+     * Return a MemoryRecords containing a single RecordBatch containing multiple Records.
+     * The batch will use the current magic.
```

Comment:
you've said it will use the current magic, but that's not what the method impl does (it uses the default).

Reply:
Hmm, the default == current, this was a way to make that consistent over the methods. Though I take your point since the default constant is public. WDYT @tombentley

Reply:
I've made the consts private for now.